bundles/core/src/test/java/com/adobe/cq/commerce/core/components/client/MagentoGraphqlClientTest.java

```diff
@@ -59,7 +59,17 @@
 public class MagentoGraphqlClientTest {
 
     private static final ValueMap MOCK_CONFIGURATION = new ValueMapDecorator(ImmutableMap.of("cq:graphqlClient", "default", "magentoStore",
-        "my-store"));
+        "my-store", "httpHeaders", new String[] { "customHeader-1=value1", "customHeader-2=value2", "customHeader-3==value3=3=3=3=3",
+            "Authorization=099sx8x7v1" }));
+
+    private static final List<Header> expectedHeaders = new ArrayList<>();
+
+    static {
+        expectedHeaders.add(new BasicHeader("customHeader-1", "value1"));
+        expectedHeaders.add(new BasicHeader("customHeader-2", "value2"));
+        expectedHeaders.add(new BasicHeader("customHeader-3", "=value3=3=3=3=3"));
+        expectedHeaders.add(new BasicHeader("Authorization", "099sx8x7v1"));
```

Comment:
I think you forgot about the filtering :)

Reply:
Yes, I did. Adding it now :)